PSICompile/ILCodeGen.cs

```diff
@@ -66,8 +66,24 @@ public class ILCodeGen : Visitor {
       }
       if (w.NewLine) Out ("    call void [System.Console]System.Console::WriteLine ()");
    }
-   
-   public override void Visit (NIfStmt f) => throw new NotImplementedException ();
+
+   public override void Visit (NIfStmt f) {
+      string lab1 = NextLabel (), lab2 = NextLabel (), lab3 = NextLabel (), lab4 = NextLabel ();
+      Out ($"  {lab1}:");
+      f.Condition.Accept (this);
+      if (f.IfPart != null) {
+         Out ($"    brfalse {lab3}");
+         Out ($"  {lab2}:");
+         f.IfPart.Accept (this);
+         Out ($"    br {lab4}");
+      } else {
+         Out ($"    brfalse {lab4}");
+      }
+      Out ($"  {lab3}:");
+      f.ElsePart?.Accept (this);
+      Out ($"  {lab4}:");

```

Comment:
It's possible to rewrite this so you need only 2 labels (if the statement has an **else** part), or just one label (if there is no **else** part). You definitely don't need 4 labels. Also, f.IfPart will never be null - that's not possible by Pascal language rules.